tools/tests/form-control/test.js

```diff
@@ -427,21 +450,80 @@ function setFormsTestCases() {
 	}
 }
 
+function setFormTestCasesWithInteraction() {
+	for (const formData of forms) {
+		if (!formData.submitDataInteraction) continue; // Skipping PREVENT_SUBMIT_FORM
+
+		const formTestName = formData.testName;
+
+		describe(formTestName, () => {
+			beforeAll(function (done) {
+				setupFormControlPage(formData).then(done).catch(err => console.error(err));
+			});
+
+			it('Should be created', () => {
+				assert(document.querySelector('gameface-form-control').id === formData.id, `The id of the form is not ${formData.id}.`);
+			});
+
+			it('Should be interacted, submitted and form data should be as expected.', async () => {
+				// Interact with elements
+				const formElement = document.querySelector('gameface-form-control');
+
+				switch (formTestName) {
+					case formsTestNames.LOGIN_FORM:
+						click(document.querySelectorAll('dropdown-option')[1]);
+						break;
+					case formsTestNames.REGISTER_FORM:
+						click(document.querySelectorAll('radio-button')[1]);
+						click(document.querySelector('gameface-checkbox'));
+						const rangeslider = document.querySelector('gameface-rangeslider');
+						const { x, width } = document.querySelector('.horizontal-rangeslider').getBoundingClientRect();
+						rangeslider.onMouseDown({ clientX: x + width });
+						break;
+					case formsTestNames.CHECKBOXES_FORM:
+						const checkboxes = Array.from(document.querySelectorAll('gameface-checkbox'));
+						checkboxes.forEach((checkbox) => click(checkbox));
+						break;
+					case formsTestNames.RADIO_FORM:
+						click(document.querySelectorAll('radio-button')[1]);
+						break;
+					case formsTestNames.SWITCH_FORM:
+						const switches = Array.from(document.querySelectorAll('gameface-switch'));
+						switches.forEach((switchElement) => switchElement.onClick());
+						break;
+					case formsTestNames.DROPDOWN_FORM:
+						const dropdowns = document.querySelectorAll('gameface-dropdown');
+						const multipleSelectOptionTwo = dropdowns[4].querySelectorAll('dropdown-option')[1];
+						click(dropdowns[0].querySelectorAll('dropdown-option')[1]);
+						click(dropdowns[1].querySelectorAll('dropdown-option')[1]);
+						click(dropdowns[2].querySelectorAll('dropdown-option')[1]);
+						click(dropdowns[3].querySelectorAll('dropdown-option')[1]);
+						multipleSelectOptionTwo.onClick({ target: multipleSelectOptionTwo, ctrlKey: true });
+						break;
+				}
+
+				// Submit
+				const submitButton = document.querySelector('[type="submit"]');
+				click(submitButton);
+				await waitServerResponse();
+				// Check data
+				await createAsyncSpec(() => {
```

Comment:
`createAsyncSpec` has a `try...catch` block within the Promise in the callback in `waitForStyles`:
```js
function createAsyncSpec(callback = () => {}, frames = DEFAULT_FRAMES_TO_WAIT) {
    return new Promise((resolve, reject) => {
        waitForStyles(() => {
            try {
                callback();
                resolve();
            } catch (error) {
                reject(error);
            }
        }, frames);
    });
}
```
If in the test we set `const data = 'wrong';`, the test will be marked as failed and this will be logged:
![image](https://user-images.githubusercontent.com/21058220/146775969-a6322e62-f802-4c5c-afa9-d2bfde3712e7.png)